ksqldb-execution/src/main/java/io/confluent/ksql/execution/codegen/CodeGenSpec.java

```diff
@@ -125,7 +124,9 @@ void addFunction(final FunctionName functionName, final Kudf function) {
 
     void addStructSchema(final CreateStructExpression struct, final Schema schema) {
       final String structSchemaName = CodeGenUtil.schemaName(structSchemaCount++);
-      structToSchemaName.put(struct, structSchemaName);
+      if (!structToSchemaName.containsKey(struct)) {
```

Comment:
I think we should flip the semantics of the if statement here and instead return early if the `structToSchemaName` already contains the entry - that way we wont be incrementing `structSchemaCount` or adding the unnecessary argument to the `argumentBuilder`.